Replace date-range-picker with a self-contained directive.

stephenwan opened the pull request:
As requested by Niclas, we have wrapped the date-range-picker into a directive with simple interface. 

It has a slight dependency on `angularMoment` for formatting the date string in the date-picker-header, but could be removed upon request.

This directive handles two types of validation by itself. 
1. ng-invalid-order   :  the start date is greater than the end date.
2. ng-invalid-empty  :  the start date or the end date is not set.

Error messages with translation will be provided automatically, but can be muted upon user's need. In the later case, the user can watch the `errors` from the binding to provide customized error messages by himself.

blandine commented:
Good to see new stuff!
Is it possible to add angular moment as a npm dependency ?
I will push the test environment with karma, please add some tests for this directive.

stephenwan commented:
Hi @bladine , I have updated the pr based on your suggestions.

blandine commented:
Thanks @stephenwan 
I've seen this empty file : js/date-range-picker/date-range-picker.directive.js~ if you could delete it.
About angular and angular material you upgraded to the latest version, is it on purpose due to the future upgrade in the project ? Or can we keep the old version for now?

stephenwan commented:
Hi @blandine, I have removed the *~ file. It was mistakenly committed before I added that pattern into .gitignore, and went unnoticed thereafter. About the versions for angular and angular-material, it actually doesn't matter for my pr. I did the naive ``` npm install *** --save ``` without thinking about compatibility. Could you help me to adjust the package.json with the versions you prefer? Thanks.

blandine commented on package.json:
```diff
@@ -9,7 +9,14 @@
   "license": "ISC",
   "dependencies": {
-    "bootstrap": "3.3.4"
+    "angular": "~1.3.2",
```
1.4.5

blandine commented on package.json:
```diff
@@ -9,7 +9,14 @@
   "dependencies": {
-    "bootstrap": "3.3.4"
+    "angular": "~1.3.2",
+    "angular-animate": "~1.3.2",
```
1.4.5

blandine commented on package.json:
```diff
@@ -9,7 +9,14 @@
-    "bootstrap": "3.3.4"
+    "angular": "~1.3.2",
+    "angular-animate": "~1.3.2",
+    "angular-aria": "~1.3.2",
```
1.4.5

blandine commented on package.json:
```diff
@@ -9,7 +9,14 @@
+    "angular": "~1.3.2",
+    "angular-animate": "~1.3.2",
+    "angular-aria": "~1.3.2",
+    "angular-material": "~0.8.3",
```
0.10.1

blandine commented on package.json:
```diff
@@ -9,7 +9,14 @@
+    "angular-animate": "~1.3.2",
+    "angular-aria": "~1.3.2",
+    "angular-material": "~0.8.3",
+    "angular-moment": "^0.10.3",
```
0.8.3

blandine commented on package.json:
```diff
@@ -9,7 +9,14 @@
+    "angular-aria": "~1.3.2",
+    "angular-material": "~0.8.3",
+    "angular-moment": "^0.10.3",
+    "angular-route": "^1.3.2",
```
1.4.5

blandine commented on package.json:
```diff
@@ -9,7 +9,14 @@
+    "angular-material": "~0.8.3",
+    "angular-moment": "^0.10.3",
+    "angular-route": "^1.3.2",
+    "bootstrap": "3.3.4",
```
3.2.0

blandine commented on package.json:
```diff
@@ -9,7 +9,14 @@
+    "angular-moment": "^0.10.3",
+    "angular-route": "^1.3.2",
+    "bootstrap": "3.3.4",
+    "moment": "^2.10.6"
```
2.8.3

blandine commented:
hi @stephenwan I put the same versions as in the project as comments, so you can update easily your commit. Btw always keep a static version of the different modules.

thanks for the changes, I'm ready to merge after this :)

stephenwan commented:
Thanks @blandine ! They are not in the correct versions.
